test/capi/capi_sarray.cxx

```diff
@@ -6,42 +6,1581 @@
+
+    turi::gl_sarray combined_gl_output = g1.append(g2);
+
+    TS_ASSERT((combined_gl_output == combined_output->value).all());
+
+    tc_flex_list_destroy(fl1);
+    tc_flex_list_destroy(fl2);
+
+    tc_sarray_destroy(sa1);
+    tc_sarray_destroy(sa2);
+
+    TS_ASSERT(error == NULL);
+  };
+
+  //TODO
+  void test_tc_sarray_unique(){
+    tc_error* error = NULL;
+
+    std::vector<double> v1 = {1, 2, 4.5, 9, 389, 23};
+
+    tc_flex_list* fl1 = make_flex_list_double(v1);
+    tc_sarray* sa1 = tc_sarray_create_from_list(fl1, &error);
+
+    turi::flex_list lst1;
+
+    for (auto it = v1.begin(); it!=v1.end(); ++it) {
+        lst1.push_back(*it);
+    }
+
+    turi::gl_sarray g1(lst1);
+
+    TS_ASSERT(error == NULL);
+
+    TS_ASSERT((g1.unique() == tc_sarray_unique(sa1, &error)->value).all());
+
+    tc_sarray_destroy(sa1);
+    TS_ASSERT(error == NULL);
+  };
+
+  void test_tc_sarray_is_materialized(){
+    tc_error* error = NULL;
+
+    std::vector<double> v1 = {1, 2, 4.5, 9, 389, 23};
+
+    tc_flex_list* fl1 = make_flex_list_double(v1);
+    tc_sarray* sa1 = tc_sarray_create_from_list(fl1, &error);
+
+    turi::flex_list lst1;
+
+    for (auto it = v1.begin(); it!=v1.end(); ++it) {
+        lst1.push_back(*it);
+    }
+
+    turi::gl_sarray g1(lst1);
+
+    TS_ASSERT(error == NULL);
+
+    tc_sarray* combined_output = tc_sarray_sample(sa1, 0.8, 2, &error);
+
+    turi::gl_sarray combined_gl_output = g1.sample(0.8, 2);
+
+    TS_ASSERT((combined_output->value == combined_gl_output).all());
+
+    TS_ASSERT((tc_sarray_is_materialized(combined_output, &error) == combined_gl_output.is_materialized()));
+
+
+    tc_flex_list_destroy(fl1);
+    tc_sarray_destroy(sa1);
+    TS_ASSERT(error == NULL);
+  };
+
+  void test_tc_sarray_materialize(){
+    tc_error* error = NULL;
+
+    std::vector<double> v1 = {1, 2, 4.5, 9, 389, 23};
+
+    tc_flex_list* fl1 = make_flex_list_double(v1);
+    tc_sarray* sa1 = tc_sarray_create_from_list(fl1, &error);
+
+    turi::flex_list lst1;
+
+    for (auto it = v1.begin(); it!=v1.end(); ++it) {
+        lst1.push_back(*it);
+    }
+
+    turi::gl_sarray g1(lst1);
+
+    TS_ASSERT(error == NULL);
+
+    tc_sarray* combined_output = tc_sarray_sample(sa1, 0.8, 2, &error);
+
+    turi::gl_sarray combined_gl_output = g1.sample(0.8, 2);
+
+    TS_ASSERT((combined_output->value == combined_gl_output).all());
+
+    TS_ASSERT((tc_sarray_is_materialized(combined_output, &error) == combined_gl_output.is_materialized()));
+
+    tc_sarray_materialize(combined_output, &error);
+    combined_gl_output.materialize();
+
+    TS_ASSERT((tc_sarray_is_materialized(combined_output, &error) == combined_gl_output.is_materialized()));
+
+    tc_flex_list_destroy(fl1);
+    tc_sarray_destroy(sa1);
+    TS_ASSERT(error == NULL);
+  };
+
+};
+
+
+BOOST_FIXTURE_TEST_SUITE(_capi_test_sarray, capi_test_sarray)
+BOOST_AUTO_TEST_CASE(test_sarray_double) {
+  capi_test_sarray::test_sarray_double();
```

Comment:
This is fine for now, but we could put the actual test code inside here.  It was separated out for historical reasons....